### PR TITLE
[Sharepoint] Support multiple file sizes

### PR DESCRIPTION
**Pull request by vidok**
This PR adds support for multiple file sizes for better Sharepoint Online/Server simulation.

if `DATA_SIZE` environment variable is passed then only one size will be used.

## Related Pull Requests

* https://github.com/elastic/enterprise-search-team/issues/4332

### CODE REVIEW

**Review by timgrein — APPROVED** [review]
One small nit around readability, otherwise LGTM

File: connectors/sources/tests/fixtures/sharepoint/fixture.py [hunk 1]
```diff
@@ -13,14 +13,60 @@
...
+
+# Generate range distribution (e.g. [range(0,2), range(2, 15), ...])
+DISTRIBUTION_RANGES = {}
+i = 0
+for k, v in FILE_SIZES_DISTRIBUTION.items():
+    DISTRIBUTION_RANGES[k] = range(i, i + v)
+    i = i + v
+
+# Generate data for different sizes
+for k, _v in FILE_SIZES_DISTRIBUTION.items():
```
Comment (tarekziade) [inline connectors/sources/tests/fixtures/sharepoint/fixture.py:53]: if you don't use the value you can just iterate on `.keys()`

**Review by tarekziade — APPROVED** [review]
just one nit.

File: connectors/sources/tests/fixtures/sharepoint/fixture.py (hunk 1 above)
Comment (tarekziade) [reply connectors/sources/tests/fixtures/sharepoint/fixture.py:53]:
```python
>>> map = {'a':1, 'b':2}
>>> for k in map:
...     print(k)
...
a
b
```

```python
>>> for k in map.keys():
...     print(k)
...
a
b
```

File: connectors/sources/tests/fixtures/sharepoint/fixture.py [hunk 2]
```diff
@@ -13,14 +13,60 @@
...
+# if DATA_SIZE is passed then use only one file size
+if DATA_SIZE and DATA_SIZE in SIZES:
+    FILE_SIZES_DISTRIBUTION = {DATA_SIZE: 100}
+else:
+    FILE_SIZES_DISTRIBUTION = {
+        "extra_small": 70,
+        "small": 0,
+        "medium": 10,
+        "large": 5,
+    }  # sum has to be 100
```
Comment (wangch079) [inline connectors/sources/tests/fixtures/sharepoint/fixture.py:40]: 70+0+10+5=85?

**Review by wangch079 — APPROVED** [review]
LGTM, assuming it's not a strict requirement to have the exact distribution

**Comment by vidok** [comment]
> LGTM, assuming it's not a strict requirement to have the exact distribution

Yes, this is right. It's not gonna be used for the current e2e test.

**Pull request merged**